src/NServiceBus.Metrics/Extensions.cs

```diff
@@ -19,14 +20,18 @@ public static bool TryGetTimeSent(this ReceivePipelineCompleted completed, out D
 
     public static bool TryGetMessageType(this ReceivePipelineCompleted completed, out string ProcessedMessageType)
     {
-        var headers = completed.ProcessedMessage.Headers;
+        return completed.ProcessedMessage.Headers.TryGetMessageType(out ProcessedMessageType);
+    }
+
+    public static bool TryGetMessageType(this IReadOnlyDictionary<string, string> headers, out string processedMessageType)
+    {
         string enclosedMessageType;
         if (headers.TryGetValue(Headers.EnclosedMessageTypes, out enclosedMessageType))
         {
-            ProcessedMessageType = enclosedMessageType;
+            processedMessageType = enclosedMessageType;
             return true;
         }
-        ProcessedMessageType = "Undefined";
+        processedMessageType = "Undefined";
```

Comment:
Why not just `default(string)` or `null`? Putting a real string in here might make it difficult to tell if a probe event related to a message type or not (you'll need to do a string comparison against a magic string).